nanoFramework.Json/JsonProperty.cs

```diff
@@ -6,19 +6,38 @@
 
 namespace nanoFramework.Json
 {
-    internal class JsonProperty : JsonToken
+    /// <summary>
+    /// Represents single JSON key value pair.
+    /// </summary>
+    public sealed class JsonProperty : JsonToken
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JsonProperty" /> class.
+        /// </summary>
         public JsonProperty()
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JsonProperty" /> class.
+        /// </summary>
+        /// <param name="name">JSON property key.</param>
+        /// <param name="value">JSON property value.</param>
         public JsonProperty(string name, JsonToken value)
         {
             Name = name;
             Value = value;
         }
 
+        /// <summary>
+        /// Gets or sets key of JSON property.
+        /// </summary>
+
         public string Name { get; set; }
```

Comment:
Do we need `Name` getter to be public? 
Same goes for `Value`...

I did a quick check on the accessors and haven't sppoted any need to set both outside the constructor.

Reply:
I agree with "set" they don't have to be public. 
But for gets I would leave them as public. It may be handy when trying to deserialize interface type to class.

Reply:
Done.,